apps/cms.c

```diff
@@ -913,12 +918,24 @@ int cms_main(int argc, char **argv)
             goto end;
     }
 
-    if ((flags & CMS_BINARY) == 0)
-        warn_binary(infile);
-    in = bio_open_default(infile, 'r',
-                          binary_files ? FORMAT_BINARY : informat);
-    if (in == NULL)
-        goto end;
+    if (digesthex != NULL) {
+        if (infile != NULL ||
+            operation != SMIME_SIGN ||
+            !(flags & CMS_DETACHED)) {
+            BIO_printf(bio_err, "Invalid use of -digest\n");
+            goto end;
+        }
+        digestbin = OPENSSL_hexstr2buf(digesthex, &digestlen);
+        if (digestbin == NULL)
+            goto end;
+    } else {
+        if ((flags & CMS_BINARY) == 0)
```

Comment:
`!(flags & ...)` above and `(flags & ...) == 0` here.  Consistency is good.

Reply:
I've changed to `!(flags & ...)` which seems more frequent. It doesn't seem to be consistent within this file though. (These are old lines that I wrapped in an else block.)